dali/operators/generic/slice/slice_attr.h

```diff
@@ -33,13 +35,30 @@
 
 namespace dali {
 
+namespace detail {
+
+enum class AxesMode : uint8_t {
+  Axes, AxisNames, AllAxes
+};
+
+/**
+ * @brief Gets a small vector with the final axes for a particular sample, given the current config
```

Comment:
I'm not sure if the output vector is really small.

```suggestion
 * @brief Gets a SmallVector with the final axes for a particular sample, given the current config
```